src/iocore/cache/unit_tests/stub.cc

```diff
@@ -27,6 +27,29 @@
 
 #include "proxy/HttpAPIHooks.h"
 
+#include "tscore/BaseLogFile.h"
+#include "tscore/Diags.h"
+
+class InitDiags
+{
+public:
+  InitDiags() : base_log_file{new BaseLogFile("stderr")}
+  {
+    DiagsPtr::set(new Diags("cache unit test", "*" /* tags */, "" /* actions */, this->base_log_file));
+  }
+
+  ~InitDiags() { delete this->base_log_file; }
+
+private:
+  BaseLogFile *base_log_file;
+};
+
+// The FetchSM class allocator below uses the diags in its constructor, so we
+// force them to be initialized prior to that. This gets "leaked" at shutdown
+// to ensure that the log file remains valid if accessed by other threads
+// during static object destruction.
+InitDiags const *g_init_diags_dummy = new InitDiags{};
```

Comment:
This is a leak isn't it?  Why not just:
```
namespace
{
InitDiags init_diags_dummy;
}
```

Reply:
That calls could result in a use after three of the log handle, which I think is worse. I'm considering suppressing the ASan report, but I don't like that either. There's really no way to make this work with the current design.

Reply:
I think a better way to fix this is to have the allocator wait for the diags to become ready, but we need to make sure this can't cause a deadlock.

Reply:
The crash this PR solves would not happen if https://github.com/apache/trafficserver/pull/11464 got merged.  (Maybe someday @zwoop  will come out of his review coma.)  The only remaining issue will be that there will be no way to enable debug trace output during static initialization.